handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -362,51 +132,6 @@ protected void replaceHandler(ChannelHandlerContext ctx, String hostname, SslCon
         }
```

Comment:
> somebody who extends AbstractSniHandler is probably comfortable with the state management and is doing it on purpose.

My concern is mostly with the replacing mechanisms and not leaking ... we can always add it later too though.

> If replaceHandler(...) throws an Exception we'll leak the SslContext if I'm not wrong?! 

`replaceHandler` takes ownership of the `sslContext`. We should clarify this in the javadocs since this method is not final.